### PR TITLE
Add docs on getting paid for contributions

### PR DESCRIPTION
**Pull request by sethmlarson**
:tada: Let's pay some open source contributors! I've created a list of guidelines, let me know what you think.

I can't decide if we want to predetermine amounts we're going to pay for issues being completed or whether we should move to labeling issues as "Paid $$$" without the amounts and then ask contributors to calculate the time spent themselves?

### CODE REVIEW

File: docs/contributing.rst [hunk 1]
```diff
@@ -94,6 +94,32 @@ further parameterize pytest for local testing.
...
 <https://docs.pytest.org/en/stable/usage.html#stopping-after-the-first-or-n-failures>`_.
 
+Getting paid for your contributions
+-----------------------------------
+
+urllib3 has a `pool of money hosted on OpenCollective <https://opencollective.com/urllib3#category-BUDGET>`_
+which the team uses to pay contributors for their work. **That could be you, too!** If you close an issue
+that is marked with the `"Paid $$$" label <https://github.com/urllib3/urllib3/issues?q=is%3Aissue+is%3Aopen+label%3A%22Paid%20%24%24%24%22>`_ then you're eligible to be paid for your work.
+
+- Ensure that you're able to receive funds from OpenCollective.
```
Comment (nateprewitt) [inline docs/contributing.rst:104]: Does OpenCollective have any information about eligibility or setup we could link to?

**Review by nateprewitt — APPROVED** [review]
Two minor comments, otherwise looks good :shipit:

**Review by pquentin — CHANGES_REQUESTED** [review]
Thank you for putting this out there! I like that we recommend $50 per hour.

File: docs/contributing.rst [hunk 2]
```diff
@@ -94,6 +94,32 @@ further parameterize pytest for local testing.
 For all valid arguments, check `the pytest documentation
 <https://docs.pytest.org/en/stable/usage.html#stopping-after-the-first-or-n-failures>`_.
 
+Getting paid for your contributions
+-----------------------------------
+
+urllib3 has a `pool of money hosted on OpenCollective <https://opencollective.com/urllib3#category-BUDGET>`_
+which the team uses to pay contributors for their work. **That could be you, too!** If you close an issue
+that is marked with the `"Paid $$$" label <https://github.com/urllib3/urllib3/issues?q=is%3Aissue+is%3Aopen+label%3A%22Paid%20%24%24%24%22>`_ then you're eligible to be paid for your work.
```
[the remark this answers is not in the document]
Comment (shazow) [reply docs/contributing.rst:102]: Also the current link in the PR results in 0 matches.

**Comment by sethmlarson** [comment]
Okay I've fixed "Open Collective", linked to another part of their FAQ for eligibility, mentioned contract/tax restrictions being a thing. Most important change is switching to a "predetermined amount" for work (what we "have" setup currently).

If we go this way I'd like to boost the amounts for current issues. Should we go with small/medium/large and $100/$300/$500, I'm guessing some of our $300 issues will likely take longer than 6 hours.

**Review by shazow — APPROVED** [review]

**Review by nateprewitt — APPROVED** [review]

**Review by pquentin — APPROVED** [review]

**Pull request merged**